client/extensions/woocommerce/state/site/status/wc-api/test/selectors.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getLastApiError,
+	getLastApiErrorCode,
+	getLastApiErrorMessage,
+} from '../selectors';
+
+describe( 'selectors', () => {
+	let stateWithoutError;
+	let stateWithError;
+
+	beforeEach( () => {
+		stateWithoutError = {
+			extensions: {
+				woocommerce: {
+					site: {
+						1337: {
+							status: {
+								wcApi: {
+								},
+							},
+						},
+					},
+				},
+			},
+		};
+
+		stateWithError = {
+			extensions: {
+				woocommerce: {
+					site: {
+						1337: {
+							status: {
+								wcApi: {
+									error: {
+										data: {
+											code: 404,
+											message: 'Not found',
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		};
+	} );
+
+	describe( 'getLastApiError', () => {
+		it( 'should not provide an error if none is logged.', () => {
+			expect( getLastApiError( stateWithoutError, 1337 ) ).to.not.exist;
+		} );
+
+		it( 'should not provide an error if no site data exists.', () => {
+			expect( getLastApiError( stateWithoutError, 1338 ) ).to.not.exist;
+		} );
+
+		it( 'should provide last error if it exists.', () => {
+			const err = getLastApiError( stateWithError, 1337 );
+			expect( err ).to.exist;
+			expect( err.code ).to.equal( 404 );
+			expect( err.message ).to.equal( 'Not found' );
+		} );
+	} );
+
+	describe( 'getLastApiErrorCode', () => {
+		it( 'should not provide an error code if none is logged.', () => {
+			expect( getLastApiErrorCode( stateWithoutError, 1337 ) ).to.not.exist;
+		} );
+
+		it( 'should not provide an error code if no site data exists.', () => {
+			expect( getLastApiErrorCode( stateWithoutError, 1338 ) ).to.not.exist;
+		} );
+
+		it( 'should match output from getLastApiError.', () => {
+			const err = getLastApiError( stateWithError, 1337 );
+			expect( getLastApiErrorCode( stateWithError, 1337 ) ).to.equal( err.code );
```

Comment:
This test doesn't guarantee that you get the code from state, only that both getLastApiError and getLastApiErrorCode return the same value.